ksqldb-engine/src/test/java/io/confluent/ksql/structured/SchemaKGroupedStreamTest.java

```diff
@@ -164,7 +165,7 @@ public void shouldBuildStepForAggregate() {
             ExecutionStepFactory.streamAggregate(
                 queryContext,
                 schemaGroupedStream.getSourceStep(),
-                Formats.of(keyFormatInfo, valueformatInfo, SerdeFeatures.of(), SerdeFeatures.of()),
+                InternalFormats.of(keyFormat, valueformatInfo),
```

Comment:
Let's not update these tests to use `InternalFormats.of()` -- the purpose of these tests is to (indirectly) verify that `InternalFormats.of()` behaves as expected in all these different cases. (There's no dedicated unit-test coverage for `InternalFormats.of()` AFAIA.) In fact, this could be a good place for you to add unit test coverage for your case (schema ID) as well.

Reply:
https://github.com/confluentinc/ksql/blob/master/ksqldb-engine/src/main/java/io/confluent/ksql/structured/SchemaKGroupedStream.java#L97
The SchemaKGroupedStream calls the `streamAggregate` method with `InternalFormats.of()` method and therefore I updated the test to reflect the same.

Reply:
Right but if you look at the different tests that you've modified in this file, the assertions are now all the same even though they're meant to be different test cases. These changes would be fine if we had separate unit tests for `SchemaKGroupedStream` and `InternalFormats` but we don't. As the test situation currently stands, these changes have removed test coverage for `InternalFormats`.

Reply:
Got it. Thanks